client/main.cpp

```diff
@@ -406,6 +411,81 @@ int boinc_main_loop() {
     return finalize();
 }
 
+//
+//
+//
+int connect_sock () {
+
+    int portno = 31416;
+    struct sockaddr_in serveraddr;
+    struct hostent *server;
+    char *hostname = "127.0.0.1";
+    int sockfd;
+
+    server = gethostbyname(hostname);
+
+    sockfd = socket(AF_INET, SOCK_STREAM, 0);
+    serveraddr.sin_family = AF_INET;
+    serveraddr.sin_port = htons(portno);
+    serveraddr.sin_addr = *((struct in_addr *)server->h_addr);
+    bzero(&(serveraddr.sin_zero), 8);
+
+    int n = connect(sockfd, (struct sockaddr *) &serveraddr, sizeof(struct sockaddr));
+    if (n < 0) {
+       printf("The connect error is : %s", strerror(errno));
+    }
+    return sockfd;
+}
+//
+//
+//
+
+//
+//
+//
+void init_websocket() {
+
+    std::vector<std::thread *> threads(/*std::thread::hardware_concurrency()*/1);
```

Comment:
There is no c++11 support on Windows client now and this function can fail while building on Visual Studio 2010. I know that we plan to support Visual Studio 2017 but there is no plans (as far as I know) to drop support of Visual Studio 2010. If we still want to support Visual Studio 2010 this must be compiled using it (unfortunately, current AppVeyor script use Visual Studio 2013 for building) or use boinc own implementation of threads.